_fr/tour/abstract-type-members.md

```diff
@@ -2,11 +2,80 @@
 layout: tour
 title: Abstract Type Members
 partof: scala-tour
-
-num: 21
-
+num: 25
 language: fr
-
 next-page: compound-types
 previous-page: inner-classes
+topics: abstract type members
+prerequisite-knowledge: variance, upper-type-bound
+
+redirect_from: 
+  - "/tutorials/tour/abstract-types.html"
+  - "/tour/abstract-types.html"
+
 ---
+
+Les types abstraits, tels que les traits et les classes abstraites, peuvent avoir des membres de type abstrait.
```

Comment:
```suggestion
Les types abstraits, tels que les traits et les classes abstraites, peuvent avoir des membres type abstrait.
```
"type member" refers to the notion of a member that *is* a type; not a thing that is a member *of* a type. So in French we should use "membre type abstrait". For the plural, it should be "membres type abstrait" because "type abstrait" is a compound noun used as an adjective. That is invariant, like "bleu roi" is invariant in "des fleurs bleu roi".